symengine/subs.h

```diff
@@ -169,6 +169,34 @@ class SubsVisitor : public BaseVisitor<SubsVisitor>
         result_ = x.create(v);
     }
 
+    void bvisit(const ImageSet &x)
+    {
+        RCP<const Basic> s = apply(x.get_symbol());
+        RCP<const Basic> expr = apply(x.get_expr());
+        auto bs_ = apply(x.get_baseset());
+        if (not is_a_Set(*bs_))
+            throw SymEngineException("expected an object of type Set");
+        RCP<const Set> bs = rcp_static_cast<const Set>(bs_);
+        if (s == x.get_symbol() and expr == x.get_expr()
+            and bs == x.get_baseset()) {
+            result_ = x.rcp_from_this();
+        } else {
+            result_ = x.create(s, expr, bs);
+        }
+    }
+
+    void bvisit(const Union &x)
+    {
+        set_set v;
+        for (const auto &elem : x.get_container()) {
+            auto a = apply(elem);
+            if (not is_a_Set(*a))
```

Comment:
Not sure about this, but what if I pass `Union(x, y) -> subs({x : z})` ? It will throw, which it shouldn't right?